src/main/java/org/checkerframework/checker/builder/TypesafeBuilderAnnotatedTypeFactory.java

```diff
@@ -107,115 +117,85 @@ public Void visitMethodInvocation(
 
       return super.visitMethodInvocation(tree, type);
     }
-  }
 
-  /**
-   * The qualifier hierarchy is responsible for lub, glb, and subtyping between qualifiers without
-   * declaratively defined subtyping relationships, like our @CalledMethods annotation.
-   */
-  private class TypesafeBuilderQualifierHierarchy extends MultiGraphQualifierHierarchy {
-    public TypesafeBuilderQualifierHierarchy(
-        final MultiGraphQualifierHierarchy.MultiGraphFactory factory) {
-      super(factory);
-    }
-
-    @Override
-    public AnnotationMirror getTopAnnotation(final AnnotationMirror start) {
-      return TOP;
+    private boolean hasReturnsReceiver(MethodInvocationTree tree) {
+      // Check to see if the @ReturnsReceiver annotation is present
+      Element element = TreeUtils.elementFromUse(tree);
+      if (getDeclAnnotation(element, ReturnsReceiver.class) != null) {
+        return true;
+      }
+      return isAutoValueBuilderSetter(element);
     }
 
-    /**
-     * GLB in this type system is set union of the arguments of the two annotations, unless one of
-     * them is bottom, in which case the result is also bottom.
-     */
-    @Override
-    public AnnotationMirror greatestLowerBound(
-        final AnnotationMirror a1, final AnnotationMirror a2) {
-      if (AnnotationUtils.areSame(a1, BOTTOM) || AnnotationUtils.areSame(a2, BOTTOM)) {
-        return BOTTOM;
+    private boolean isAutoValueBuilderSetter(Element element) {
+      MethodTree methodTree = (MethodTree) declarationFromElement(element);
```

Comment:
I could not figure out how to insert declaration annotations in a type factory.  And making `@ReturnsReceiver` a type use annotation seemed wrong.  So I added inline logic to treat methods as having `@ReturnsReceiver` when they are setters in AutoValue builders.  We need to make this configurable eventually.